### ISSUE TITLE
[Bug] Windows 1.20 can not encrypt system drive

### ISSUE THREAD

**Issue by nshtg** [veracrypt/VeraCrypt#181]
I tried encrypting my system drive using 1.20 but I can't get past the "Rescue Disk" part.

I tick "Skip Rescue Disk verification" click on continue and am greeted with:

```
---------------------------
VeraCrypt
---------------------------
Parameter incorrect.


(If you report a bug in connection with this, please include the following technical information in the bug report:
VeraCrypt::BootEncryption::CreateRescueIsoImage:2894)
---------------------------
OK   
---------------------------
```

* Windows 10 Education
* Samsung 960 EVO NVMe SSD

**Comment by idrassi** [comment]
Thank you for your report.
This issue has been fixed and it will be included in 1.21 that is coming tomorrow (see https://github.com/veracrypt/VeraCrypt/commit/33bb507fdeb0ca01725ad76f70dc00a7c154ed13
An installer for the release candidate of 1.21 is available at https://sourceforge.net/projects/veracrypt/files/VeraCrypt%20Nightly%20Builds/ if you want to do tests on your side to confirm the fix.

**Comment by vlakoff** [comment]
An error on 1.21, maybe related: #260

**Comment by jkl777** [comment]
I have te same problem with W10 1809. I had to decrypt my SSD in order to apply this update. After that I cannot encrypt system drive because this error:

Parameter incorrect.


(If you report a bug in connection with this, please include the following technical information in the bug report:
VeraCrypt::BootEncryption::CreateRescueIsoImage:3439)
---------------------------

Tried all latest VC plus beta editions. Nothing. Only error code changes.

**Comment by vlakoff** [comment]
@jkl777 Please provide the VeraCrypt version corresponding to the error message, otherwise we can't be sure what line in [CreateRescueIsoImage::CreateRescueIsoImage](https://github.com/veracrypt/VeraCrypt/blob/a20b3083b058901572953364dccbdd4cad8f3c5a/src/Common/BootEncryption.cpp#L3376) threw the exception, as the file (thus the line numbers) changes over time.

**Comment by jkl777** [comment]
Latest Stable.  1.23 Hotfix 2 x64

Tried BETA also without any results.

I had also: 3006 error and 3455 error but I don't remember versions. All were above 1.21 for sure.

**Comment by vlakoff** [comment]
1.23-Hotfix-2 is from 8 october 2018.

Digging into [history for CreateRescueIsoImage::CreateRescueIsoImage](https://github.com/veracrypt/VeraCrypt/commits/master/src/Common/BootEncryption.cpp):
* There is no `throw` on line 3439 for file version immediately preceding 8 october, i.e. of 3 september.
* But there is a `throw` on line 3439 for file version of 10 october, I guess that's the one.

Code excerpt:
https://github.com/veracrypt/VeraCrypt/blob/4fea2089589bae73d5ed41b39142caac63519d98/src/Common/BootEncryption.cpp#L3436-L3439

See: https://stackoverflow.com/questions/34318600/zip-open-gives-an-unknown-error

@idrassi `int ierr = 0;` might fix the issue :)

**Comment by vlakoff** [comment]
Mmm, not sure, as the code was working previously and `ierr` isn't read.
I'm leaving you the rest of the investigation :)

**Comment by jkl777** [comment]
1.23-Hotfix-2 is from 8 october 2018 --> 3439 error

**Comment by vlakoff** [comment]
Just to confirm: 3006 and 3455 are for the same error, on older versions.

**Comment by alt3r-3go** [comment]
@jkl777, have you by any chance tried the latest 1.24-Beta5? If not, give it a try.

**Comment by idrassi** [comment]
Duplicate of https://github.com/veracrypt/VeraCrypt/issues/441 which was fixed in 1.24-Hotfix1
